### ISSUE TITLE
Nullpointer exception with empty file name in ZipParameters

### ISSUE THREAD

**Issue by pdenooijer** [srikanth-lingala/zip4j#388]
Since [this change](https://github.com/srikanth-lingala/zip4j/compare/v2.9.0...v2.9.1#diff-100b4db33b29b6e975cd85b189f47189ef011d29acf01618ad322a19a27e1a45L75) from version 2.9.0 tot 2.9.1 the file name could still be null leading to a null pointer exception.

The validation is done in the `FileHeaderFactory`, but this is called from the `initializeAndWriteFileHeader`. As this is now after the directory check, this will never trigger:
```
  private String validateAndGetFileName(String fileNameInZip) throws ZipException {
    if (!Zip4jUtil.isStringNotNullAndNotEmpty(fileNameInZip)) {
      throw new ZipException("fileNameInZip is null or empty");
    } else {
      return fileNameInZip;
    }
  }
```

Problematic code, 2e line is the problem when file name is null:
```
ZipParameters clonedZipParameters = new ZipParameters(zipParameters);
if (isZipEntryDirectory(zipParameters.getFileNameInZip())) {
  clonedZipParameters.setWriteExtendedLocalFileHeader(false);
  clonedZipParameters.setCompressionMethod(CompressionMethod.STORE);
  clonedZipParameters.setEncryptFiles(false);
}
initializeAndWriteFileHeader(clonedZipParameters);
```

There are multiple solutions to solve this. Moving the validation to an earlier point or making sure there is no null pointer exception by checking if it's not null.

**Comment by srikanth-lingala** [comment]
Thanks for the elaborate writeup @pdenooijer. I will fix it.

**Comment by srikanth-lingala** [comment]
Fixed in v2.10.0 released today
